include/seqan3/io/sequence_file/input.hpp

```diff
@@ -819,7 +820,27 @@ class sequence_file_input
  * \{
  */
 
-//!\brief Deduction of the selected fields, the file format and the stream type.
+//!\brief Deduces the sequence input file type from the stream and the format.
+template <IStream2                           stream_type,
+          SequenceFileInputFormat            file_format>
+sequence_file_input(stream_type & stream,
+                    file_format const &)
+    -> sequence_file_input<typename sequence_file_input<>::traits_type,         // actually use the default
+                           typename sequence_file_input<>::selected_field_ids,  // default field ids.
+                           type_list<file_format>,
+                           typename std::remove_reference_t<stream_type>::char_type>;
+
+//!\overload
+template <IStream2                           stream_type,
+          SequenceFileInputFormat            file_format>
+sequence_file_input(stream_type && stream,
+                   file_format const &)
+   -> sequence_file_input<typename sequence_file_input<>::traits_type,         // actually use the default
+                          typename sequence_file_input<>::selected_field_ids,  // default field ids.
+                          type_list<file_format>,
+                          typename std::remove_reference_t<stream_type>::char_type>;
+
+//!\brief Deduces the sequence input file type from the stream, the format and the field ids.
```

Comment:
Why are new overloads needed?

Reply:
Because the test was testing wrong behavior and this guide was necessary if only the stream and the format is given.